sale_coupon_limit/models/sale_coupon_rule.py

```diff
@@ -115,11 +115,14 @@ def _compute_rule_times_used(self):
                         ]
                     )
                     continue
-                salesman_limit.rule_times_used = self.env["sale.order"].search_count(
-                    [
-                        "|",
-                        ("no_code_promo_program_ids", "in", [program["id"]]),
-                        ("code_promo_program_id", "=", [program["id"]]),
-                        ("user_id", "=", salesman_limit.rule_user_id.id),
-                    ]
+                salesman_limit.rule_times_used = len(
+                    self.env["sale.order.line"].read_group(
+                        [
+                            ("coupon_program_id", "=", [program["id"]]),
+                            ("order_id.user_id", "=", salesman_limit.rule_user_id.id),
```

Comment:
Isn't this field at line level?

Reply:
It's not a question of index, but I see that there's no `auto_join=True`, so this will be decomposed in 2 queries:

- One getting the orders with such user.
- Another for sales lines with `order_id IN ...` a a big bunch of IDs, which is not good for the performance when this list goes high.

Reply:
We can do it in two steps: order ids for that user and then order lines.

Reply:
I'm afraid the result is going to be the same. There are 2 options:

- Put auto_join=True, affecting if the are specific record rules for visibility different from header and lines.
- Put a related stored field in lines for the user.

Anyways, let's keep it for now this way, and put a comment about the performance in the line, and I fast track it.